spec/system/distribution_system_spec.rb

```diff
@@ -57,6 +62,51 @@
       expect(page).to have_selector "#distribution_line_items"
     end
 
+    context "when the delivery_method is shipped" do
+      context "when the shipping cost is negative" do
+        it "should not create distribution and show error message" do
+          visit @url_prefix + "/distributions/new"
+
+          select @partner.name, from: "Partner"
+          select @storage_location.name, from: "From storage location"
+          choose "Shipped"
+
+          # to check if shipping_cost field exist
+          expect(page.find_by_id("shipping_cost_div")).not_to be_nil
+
+          fill_in "Shipping cost", with: '-12.05'
+          fill_in "Comment", with: "Take my wipes... please"
+          fill_in "Distribution date", with: '01/01/2001 10:15:00 AM'
+
+          click_button "Save", match: :first
+
+          expect(page.find(".alert-danger")).to have_content "Sorry, we weren't able to save the distribution. Shipping cost cannot be negative!"
+        end
+      end
+
+      context "when the shipping cost is positive" do
+        it "Allows a distribution to be created" do
+          visit @url_prefix + "/distributions/new"
+
+          select @partner.name, from: "Partner"
+          select @storage_location.name, from: "From storage location"
+          choose "Shipped"
+
+          # to check if shipping_cost field exist
+          expect(page.find_by_id("shipping_cost_div")).not_to be_nil
+
+          fill_in "Shipping cost", with: '12.05'
+          fill_in "Comment", with: "Take my wipes... please"
+          fill_in "Distribution date", with: '01/01/2001 10:15:00 AM'
+
+          click_button "Save", match: :first
+
+          expect(page).to have_content "Distributions"
+          expect(page.find(".alert-info")).to have_content "created"
+        end
+      end
+    end
```

Comment:
Do we need two tests here? System tests are expensive - let's just do the happy path. We should lean on model tests for things like validations.

Reply:
removed test for negative shipping_cost scenario